### DISCUSSION TITLE
Is there a way to highlight atoms in a reaction?

### DISCUSSION THREAD

**Discussion by mikemhenry** [rdkit/rdkit#5020] [Q&A] [answered]
I haven't been able to figure out the API for highlighting atoms in a reaction. I see how to do it with `DrawMolecule` and `MolsToGridImage` but I'm not sure how to manipulate the `rdkit.Chem.rdChemReactions.ChemicalReaction` to add atom and bond highlights.

**Comment by DavidACosgrove** [comment, marked as answer]
Hi Mike,
If you want to show the atoms in each reagent and where they ended up in
the products, you can do:

rxn = AllChem.ReactionFromSmarts(
  '[CH3:1][C:2](=[O:3])[OH:4].[CH3:5][NH2:6]>CC(O)C.[Pt]>[CH3:1][C:2](=[O:3])[NH:6][CH3:5].[OH2:4]',
  useSmiles=True)
colors = [(0.3, 0.7, 0.9), (0.9, 0.7, 0.9), (0.6, 0.9, 0.3), (0.9, 0.9, 0.1)]
d = Draw.MolDraw2DSVG(900, 300)
d.DrawReaction(rxn, highlightByReactant=True, highlightColorsReactants=colors)

The third parameter highlightColorsReactants is optional.  If not used, it
will use DrawOptions().highlightColourPalette which I personally find a bit
insipid and hard to see.

And yes, sorry, the mixture of American (color) and British (colour)
spellings is correct.  I was clearly feeling overly Imperial when I wrote
the code originally.  I've grown up since!

I assume you're using Python.  It's similar in C++.


HTH,

Dave

On Fri, Feb 18, 2022 at 3:41 AM Mike Henry ***@***.***> wrote:

> I haven't been able to figure out the API for highlighting atoms in a
> reaction. I see how to do it with DrawMolecule and MolsToGridImage but
> I'm not sure how to manipulate the
> rdkit.Chem.rdChemReactions.ChemicalReaction to add atom and bond
> highlights.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/rdkit/rdkit/discussions/5020>, or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ACGF2FXWL254EPRPM4MK6KLU3W5V5ANCNFSM5OWT5G2Q>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
> You are receiving this because you are subscribed to this thread.Message
> ID: ***@***.***>
>


-- 
David Cosgrove
Freelance computational chemistry and chemoinformatics developer
http://cozchemix.co.uk

**Reply by mikemhenry** [reply]
Thanks! 
>  I was clearly feeling overly Imperial when I wrote
the code originally.  I've grown up since!

:rofl:
